src/app/complex/complex-results/complex-navigator/service/state/complex-navigator-display.service.ts

```diff
@@ -85,4 +85,56 @@ export class NavigatorStateService {
         return paramValue;
     }
   }
+
```

Comment:
We should move some of the computed variables we have in different components here, to reduce duplication and also to keep the logic about the width size here.
```
isSorted = computed(() => this.componentsSorting() !== NavigatorComponentSorting.DEFAULT);  isOrganismSorting = computed(() => this.componentsSorting() === NavigatorComponentSorting.ORGANISM);
sortingWidth = computed(() => this.isSorted() ? 20 : 0);
interactorWidth = computed(() => 230 + this.sortingWidth());
```

Then we would need to update `complex-navigator.component.html`
```
<div class="ComplexNavigator" #parent
     [class.horizontal]="complexSearch().elements.length <= 6"
     [class.sorted]="state.isSorted()"
     [style.--cols]="numberOfColumns()"
     [style.--colSize.px]="state.columnWidth()"
     [style.--sizeSpaceHolder.px]="state.spaceHolderWidth()"
     [style.--intHeader.px]="state.interactorWidth()"
     [style.--sortingGroupSize.px]="state.sortingWidth()"
>
```
And `table-interactor-column.component.html`, lines 10-12
```
                  [matTooltip]="state.isOrganismSorting() ? (range.value | species) : range.value">
                <div class="interactorNameContainer" [style.--rowspanSize]="range.length">
                  @if (state.isOrganismSorting()) {
```
And line 60
```
@if (j === 0 && state.isSorted()) {
```

We can also create a new method in this class to calculate the width, as we're using mostly variables from this class already
```
  public setColumnWidth(totalComponentWidth: number, numberOfColumns): void {
    this.columnWidth.set(Math.max(
      (totalComponentWidth - this.interactorWidth() - this.spaceHolderWidth()) / numberOfColumns,
      70
    ));
  }
```

Then, in `complex-navigator.component.ts` we can simplify the `adjustColWidth` method.
```
  @HostListener('window:resize', ['$event'])
  adjustColWidth() {
    setTimeout(() => this.state.setColumnWidth(this.parent.nativeElement.clientWidth, this.numberOfColumns()));
  }
```

And finally remove the unused computed variables from the 2 components mentioned.